stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/statement/BatchUpdateStatement.java

```diff
@@ -105,7 +107,31 @@ public Object execute() {
             }
             // handleErrorOnTransaction(context);
             // checkAndObserveSQLError(context, e.getMessage());
-            return SQLDatasourceUtils.getSQLConnectorError(e, errorMessage + ": ");
+            return SQLDatasourceUtils.getSQLDatabaseError(e, errorMessage + ": ");
+        } catch (DatabaseException e) {
+            String errorMessage = "execute batch update failed";
+            if (!isInTransaction) {
+                try {
+                    conn.rollback();
+                } catch (SQLException ex) {
+                    errorMessage += ", failed to rollback any changes happened in-between";
+                }
+            }
```

Comment:
aren't this catch block and the previous catch block the same?

Reply:
In both catch I'm calling getSQLDatabaseError method with different exception. So it is calling the overloaded getSQLDatabaseError method.